sql/src/main/java/org/apache/druid/sql/calcite/expression/builtin/ArrayContainsOperatorConversion.java

```diff
@@ -95,7 +95,7 @@ public DimFilter toDruidFilter(
     final DruidExpression leftExpr = druidExpressions.get(0);
     final DruidExpression rightExpr = druidExpressions.get(1);
 
-    if (leftExpr.isSimpleExtraction()) {
+    if (leftExpr.isSimpleExtraction() && !(leftExpr.getDruidType() != null && leftExpr.getDruidType().isArray())) {
```

Comment:
What if it's an `array_contains()` over just a normal single-valued `String` column?  Shouldn't that also match the filter, pretending that each row contains an array of size 1?

Reply:
the type shouldn't show up as array here if it isn't an array column